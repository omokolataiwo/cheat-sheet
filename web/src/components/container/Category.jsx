import React from 'react';
import Cheat from './Cheat';

const Category = ({ category, onAddToFavorite }) => {
  let { cheats } = category;
  cheats = cheats || [];
  if (!cheats.length) {
    return null;
  }

  return (
    <section className="category-section">
      <h5>{category.title}</h5>
      <div>
        {cheats.map(cheat => (
          <Cheat cheat={cheat} onAddToFavorite={onAddToFavorite} showActionBar />
        ))}
      </div>
    </section>
  );
};

export default Category;
