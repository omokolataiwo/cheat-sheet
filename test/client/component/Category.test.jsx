import React from 'react';
import { shallow } from 'enzyme';
import Category from '../../../web/src/components/container/Category';

const props = {
  category: { title: 'Initialisation', cheats: [{ description: 'init repository', line: 'git init' }] },
  onAddToFavorite: jest.fn(() => {})
};
const wrapper = shallow(
  <Category {...props} />
);

describe('Category Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ ...props, category: {} });
  });
});
