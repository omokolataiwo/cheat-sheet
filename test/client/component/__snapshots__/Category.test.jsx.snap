// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Category
    category={
      Object {
        "cheats": Array [
          Object {
            "description": "init repository",
            "line": "git init",
          },
        ],
        "title": "Initialisation",
      }
    }
    onAddToFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Initialisation
        </h5>,
        <div>
          <Cheat
            cheat={
              Object {
                "description": "init repository",
                "line": "git init",
              }
            }
            onAddToFavorite={[MockFunction]}
            showActionBar={true}
          />
        </div>,
      ],
      "className": "category-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Initialisation",
        },
        "ref": null,
        "rendered": "Initialisation",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Cheat
              cheat={
                Object {
                  "description": "init repository",
                  "line": "git init",
                }
              }
              onAddToFavorite={[MockFunction]}
              showActionBar={true}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cheat": Object {
                "description": "init repository",
                "line": "git init",
              },
              "onAddToFavorite": [MockFunction],
              "showActionBar": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Initialisation
          </h5>,
          <div>
            <Cheat
              cheat={
                Object {
                  "description": "init repository",
                  "line": "git init",
                }
              }
              onAddToFavorite={[MockFunction]}
              showActionBar={true}
            />
          </div>,
        ],
        "className": "category-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Initialisation",
          },
          "ref": null,
          "rendered": "Initialisation",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Cheat
                cheat={
                  Object {
                    "description": "init repository",
                    "line": "git init",
                  }
                }
                onAddToFavorite={[MockFunction]}
                showActionBar={true}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "cheat": Object {
                  "description": "init repository",
                  "line": "git init",
                },
                "onAddToFavorite": [MockFunction],
                "showActionBar": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
